src/common/log.rs

```diff
@@ -0,0 +1,40 @@
+use crate::google::protobuf::{value::Kind, Struct, Timestamp, Value};
+use crate::proto::app::v1::LogEntry;
+use chrono::{DateTime, FixedOffset};
+use std::collections::HashMap;
+
+pub fn config_log_entry(
+    cfg_received_datetime: DateTime<FixedOffset>,
+    err: Option<&anyhow::Error>,
+) -> LogEntry {
+    let secs = cfg_received_datetime.timestamp();
+    let nanos = cfg_received_datetime.timestamp_subsec_nanos();
+    let level = match err {
+        Some(_) => "error".to_string(),
+        None => "info".to_string(),
+    };
+    let message = match err {
+        Some(err) => format!("could not create robot from config: {err}"),
+        None => "successfully created robot from config".to_string(),
+    };
+    LogEntry {
+        host: "esp32".to_string(),
+        level,
+        time: Some(Timestamp {
+            seconds: secs,
+            nanos: nanos as i32,
+        }),
+        logger_name: "robot_server".to_string(),
+        message,
+        caller: Some(Struct {
+            fields: HashMap::from([(
+                "Defined".to_string(),
```

Comment:
what does that mean?

Reply:
I honestly don't know, but without this the logs won't show up in app because the frontend tries to access undefined properties for display